### PR TITLE
Fixes/359 type error regex url pattern

### LINKED ISSUE #537: Rendering `data` in template has different results after upgrading to 0.10.0.
I have an issue with rendering data in my product detail template after upgrading to 0.10.0.

I used to have the `{{ data }}` variable set up in my template to pass all the product data to javascript like so: `var PRODUCT = $.parseJSON('{{ data|rest_json|escapejs }}');`

After trying to upgrade the `{{ data }}` object has changed and now contains the entire **template_context** along side the product serialized data.

What used to be:
`{u'name': u'Trousers trend', u'url': 'http://localhost:8000/catalog/trousers-trend/', u'slug': u'trousers-trend'}`

is now:
`{u'paginator': <rest_framework.pagination.PageNumberPagination object at 0x1090d2290>, u'product': <Product: Trousers trend>, u'name': u'Trousers trend', u'url': 'http://localhost:8000/catalog/trousers-trend/', u'args': (), u'request': <rest_framework.request.Request object at 0x108bd1350>, u'edit_mode': False, u'kwargs': {u'slug': u'trousers-trend'}, u'response': <Response status_code=200, "text/html; charset=utf-8">, u'data': {...}, u'slug': u'trousers-trend', u'view': <shopit.views.product.ProductDetailView object at 0x108eafd50>}`

inside the data context variable. Also the serializer fields are accessible directly inside the context: `{{ name }}`, `{{ slug }}`...

I have managed to get around this by changing `renderers.CMSPageRenderer.render()` method.

I've changed:
`template_context = self.get_template_context(data, renderer_context)`
to:
`template_context = self.get_template_context({}, renderer_context)`

That way the serializer fields are not passed in directly to the context but only later added with:
`template_context['data'] = data`

I upgraded Django to version 1.10.7, Django CMS to 3.4.2 and django-rest-framework to 3.6.2 in the process.

Is this an issue for anyone else?

### LINKED ISSUE #539: Error with search trying tutorial
I got an error while trying tutorial - after populating the database using fixtures any manage.py execution throws:
TypeError: 'RegexURLPattern' object is not iterable

This is somehow connected to search module - after deleting "search" page from cms all works.
Tried on ubuntu, both 16.10 and clean 16.04, docker image works right too.

### PR DESCRIPTION
**Pull request by jrief**
This fixes #539  reported by @ilyaglar

Also fixes #537.

### CODE REVIEW

File: shop/rest/renderers.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         view = renderer_context['view']
         request = renderer_context['request']
         response = renderer_context['response']
-        template_context = self.get_template_context(data, renderer_context)
+        template_context = self.get_template_context(dict(data), renderer_context)
```
Comment (rfleschenberg) [inline shop/rest/renderers.py]: What is the reason for wrapping `data` in `dict()` here?

Comment (jrief) [inline shop/rest/renderers.py]: ``self.get_template_context`` conditionally returns a reference on data, hence we would modify ``data`` by modifying the ``template_context``, unless we create a copy of data.

Comment (rfleschenberg) [inline shop/rest/renderers.py]: Is this related to the topic of this PR, or is this a separate issue?

**Comment by jrief** [comment]
yes, this PR fixes #539 and #537

**Comment by jrief** [comment]
this will trigger the 0.10.1 release

**Pull request merged**